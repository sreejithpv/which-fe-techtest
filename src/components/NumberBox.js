
import React from 'react';
import PropTypes from 'prop-types';

function NumberBox({ value, onClick, isSelected, isMultipliable }) {
  return (
    <li
      className={`grid-item ${isMultipliable ? 'active' : ''} ${isSelected ? 'selected' : ''}`}
      role="button" 
      onClick={onClick}>
      {value}
    </li>
  );
}

NumberBox.propTypes = {
  value: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
  isSelected: PropTypes.bool.isRequired,
  isMultipliable: PropTypes.bool.isRequired,
};

export default React.memo(NumberBox);
