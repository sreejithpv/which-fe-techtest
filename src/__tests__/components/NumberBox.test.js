import '@testing-library/jest-dom'
import TestRenderer from 'react-test-renderer';
import NumberBox from '../../components/NumberBox';

const onClick=jest.fn();
const baseProps = {
  value: 10,
  onClick,
  isSelected: false,
  isMultipliable: false,
}
const renderFun = ( props = baseProps ) => TestRenderer.create(<NumberBox {...props} />);

describe('NumberBox Component', () => {
  test("should render snapshot with default props", () => {
    const tree = renderFun();
    expect(tree.toJSON()).toMatchSnapshot();
  });
  test("should render selected class name when isSelected", () => {
    const props = {
      ...baseProps,
      isSelected: true,
    }
    const tree = renderFun(props);
    expect(tree.toJSON()).toMatchSnapshot();
  });
  test("should render active class name when isMultipliable", () => {
    const props = {
      ...baseProps,
      isMultipliable: true,
    }
    const tree = renderFun(props);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
