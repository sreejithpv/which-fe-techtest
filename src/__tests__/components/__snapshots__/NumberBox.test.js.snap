// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberBox Component should render active class name when isMultipliable 1`] = `
<li
  className="grid-item active "
  onClick={[MockFunction]}
  role="button"
>
  10
</li>
`;

exports[`NumberBox Component should render selected class name when isSelected 1`] = `
<li
  className="grid-item  selected"
  onClick={[MockFunction]}
  role="button"
>
  10
</li>
`;

exports[`NumberBox Component should render snapshot with default props 1`] = `
<li
  className="grid-item  "
  onClick={[MockFunction]}
  role="button"
>
  10
</li>
`;
